### PR TITLE
Add the KJ-Modify RS40 PCB keyboard

### PR DESCRIPTION
**Pull request by AuditeMarlow**
## Description

Add the KJ-Modify RS40 PCB keyboard

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/kj_modify/rs40/info.json [hunk 1]
```diff
@@ -0,0 +1,77 @@
+{
+    "manufacturer": "KJ-Modify",
+    "keyboard_name": "kj_modify/rs40",
```
Comment (elpekenin) [inline keyboards/kj_modify/rs40/info.json:3]:
```suggestion
    "keyboard_name": "rs40",
```

File: keyboards/kj_modify/rs40/info.json [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true
+    },
+    "matrix_pins": {
+        "cols": ["B12", "B14", "B15", "A8", "A9", "C13", "B9", "B8", "B5", "B4", "B3", "A15"],
+        "rows": ["B13", "A10", "B1", "B10"]
+    },
+    "processor": "STM32F401",
+    "url": "",
```
Comment (elpekenin) [inline keyboards/kj_modify/rs40/info.json:20]: I think this has to link to some source where to buy/build the keyboard, perhaps the one at `README/Hardware Availability`

File: data/mappings/keyboard_aliases.hjson [hunk 3]
```diff
@@ -513,6 +513,9 @@
     "ropro": {
         "target": "kingly_keys/ropro"
     },
+    "rs40": {
+        "target": "kj_modify/rs40"
+    },
```
Comment (elpekenin) [inline data/mappings/keyboard_aliases.hjson:518]: No idea if these aliases are only meant for backwards compatibility (when a board's files get moved into a new folder) or if they can be used in a "normal" scenario like this too

**Review by elpekenin — CHANGES_REQUESTED** [review]

File: keyboards/kj_modify/rs40/info.json (hunk 1 above)
Comment (elpekenin) [reply keyboards/kj_modify/rs40/info.json:3]: I **think** it is okay

**Review by waffle87 — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**